### ISSUE TITLE
too many useState's

### ISSUE THREAD

**Issue by ghost** [the-bionic/starwars#1]
it might be better alternative to create a new hook `useAppState` based on `useReducer` and 
`useEffect`, and use this to create the app wrapper.

**Comment by patrick-emmanuel** [comment]
I think we can take it a step further with `useContext` to make the state more composable.

**Comment by ghost** [comment]
Good idea! And while you're at it, focus on the shape of your state to make it as efficient and extensible as possible.
